client/src/Attendance/Attendance.js

```diff
@@ -0,0 +1,281 @@
+  const reduceAttendanceData = data =>
+    data.reduce((acc, cv) => {
+      return {
+        ...acc,
+        ...{
+          [cv.id]: [...Array(7).keys()].map(() => ({}))
+        }
+      }
+    }, {})
+  const [attendanceData, setAttendanceData] = useState(
+    reduceAttendanceData(cases)
+  )
+  const [columnDates, setColumnDates] = useState(
+    [...Array(7).keys()].map(() => '')
+  )
+  const latestAttendanceData = useRef(attendanceData)
+  const latestColumnDates = useRef(columnDates)
+
+  const updateAttendanceData = (updates, record, i) => {
+    const newArr = latestAttendanceData.current[record?.id].map(
+      (value, index) => {
+        // this logic adds and removes fields as needed depending on whether checkin/out or an absence is selected
+        return index === i
+          ? Object.keys(updates).length === 0 ||
+            (Object.keys(value).includes('absence') &&
+              (Object.keys(updates).includes('check_in') ||
+                Object.keys(updates).includes('check_out'))) ||
+            (Object.keys(updates).includes('absence') &&
+              (Object.keys(value).includes('check_in') ||
+                Object.keys(value).includes('check_out')))
+            ? updates
+            : { ...value, ...updates }
+          : value
+      }
+    )
+    latestAttendanceData.current = {
+      ...attendanceData,
+      [record.id]: newArr
+    }
+    setAttendanceData(prevData => ({ ...prevData, [record.id]: newArr }))
+  }
+
+  const handleDateChange = (index, ds) => {
+    const updatedDates = latestColumnDates.current.map((value, i) =>
+      index === i ? ds : value
+    )
+    latestColumnDates.current = updatedDates
+    setColumnDates(updatedDates)
+  }
+
+  const generateColumns = () => {
+    let cols = []
+    for (let i = 0; i < 7; i++) {
+      cols.push({
+        dataIndex: 'date' + i,
+        key: 'date' + i,
+        width: 398,
+        // eslint-disable-next-line react/display-name
+        title: () => (
+          <DatePicker
+            disabledDate={c => c && c.valueOf() > Date.now()}
+            onChange={(_, ds) => handleDateChange(i, ds)}
+            bordered={false}
+            placeholder={t('selectDate')}
+            style={{ width: '8rem ', color: '#004A6E' }}
+          />
+        ),
+        // eslint-disable-next-line react/display-name
+        render: (_, record) => {
+          return (
+            <AttendanceDataCell
+              record={record}
+              columnIndex={i}
+              updateAttendanceData={updateAttendanceData}
+            />
+          )
+        }
+      })
+    }
+
+    return [
+      {
+        title: t('childNameCap'),
+        dataIndex: 'name',
+        width: 250,
+        key: 'name',
+        // eslint-disable-next-line react/display-name
+        render: (_, record) => {
+          return (
+            <div>
+              <p className="text-lg mb-1">
+                {record.childName || record.child.childName}
```

Comment:
I'm interested in this - why do we need to have this or clause?

Reply:
@katelovescode yes! this is a bit of an annoying thing due to how we transform the data to be able to be rendered the way we want it to via the ant table component. for NE we need to send in all the necessary data points associated with the child column (name, business, case number) in one object to make them renderable. this makes the data for ne and then every other dashboard that we are using in the components a bit different.